hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/MasterQuotaManager.java

```diff
@@ -284,6 +285,9 @@ public void delete() throws IOException {
         SpaceQuotaSnapshot currSnapshotOfTable =
             QuotaTableUtil.getCurrentSnapshotFromQuotaTable(masterServices.getConnection(), table);
         QuotaUtil.deleteTableQuota(masterServices.getConnection(), table);
+        // Removes the tableName from the current state in QuotaObserverChore
+        QuotaObserverChore quotaChore = ((HMaster)masterServices).getQuotaObserverChore();
```

Comment:
Maybe we need to make all the Chores available from `MasterServices` interface?

Reply:
@ndimiduk I didn't get this comment. Masterservices interface doesn't have QuotaObserverChore. Please let me know if have something wrong here.

Reply:
I'm wondering if we should update the `MasterServices` interface to include access to the chores. This kind of inline casting defeats the purpose of having the `MasterServices` interface in the first place.

Reply:
@ndimiduk I have updated the PR, PTAL.